### PR TITLE
Implement Export-WinGetPackage powershell cmdlet for `winget download`

### PR DESCRIPTION
**Pull request by ryfu-msft**
Resolves: #3972 

- Implements the `Export-WinGetPackage` cmdlet which is the powershell equivalent of `winget download` 
- Some light refactoring was done to help share arguments across install, update, download, and uninstall.
- Added an abstract base class called `InstallerSelectionCmdlet` which is responsible for filtering out installers for those commands that deal with installer selection
- Adds support for `-InstallerType` argument
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/3977)

### CODE REVIEW

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/InstallerSelectionCmdlet.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+// -----------------------------------------------------------------------------
+
+namespace Microsoft.WinGet.Client.Commands
+{
+    using System.Management.Automation;
+    using Microsoft.WinGet.Client.Commands.Common;
+    using Microsoft.WinGet.Client.PSObjects;
+
+    /// <summary>
+    /// This is the base class for all commands that select and installer from a given package.
```
Comment (msftrubengu) [inline src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/InstallerSelectionCmdlet.cs:14]: select an installer? or select and installs?

**Review by msftrubengu — DISMISSED** [review]

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/InstallerSelectionCmdlet.cs (hunk 1 above)
Comment (ryfu-msft) [reply src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/InstallerSelectionCmdlet.cs]: Fixed. Should be `select an installer`

**Review by yao-msft — APPROVED** [review]

**Review by yao-msft — APPROVED** [review]

**Pull request merged**